PWGHF/D2H/TableProducer/dataCreatorDV0Reduced.cxx

```diff
@@ -0,0 +1,393 @@
+        else
+          invMassD = candD.invMassAntiDstar() - candD.invMassD0Bar();
+        massD = MassDStar;
+        pVecD = candD.pVector();
+        secondaryVertexD[0] = candD.xSecondaryVertexD0();
+        secondaryVertexD[1] = candD.ySecondaryVertexD0();
+        secondaryVertexD[2] = candD.zSecondaryVertexD0();
+        prongIdsD[0] = candD.prong0Id();
+        prongIdsD[1] = candD.prong1Id();
+        prongIdsD[2] = candD.prongPiId();
+        v0_type = V0_type::K0s;
+        d_type = candD.signSoftPi() * D_type::Dstar;
+        massV0 = MassK0Short;
+      } else if constexpr (std::is_same<C, CandsDplusFiltered>::value) {
+        switch (DecayChannel) {
+          case DecayChannel::DsStar2ToDplusK0s:
+            invMassD = hfHelper.invMassDplusToPiKPi(candD);
+            massD = MassDPlus;
+            pVecD = candD.pVector();
+            secondaryVertexD[0] = candD.xSecondaryVertex();
+            secondaryVertexD[1] = candD.ySecondaryVertex();
+            secondaryVertexD[2] = candD.zSecondaryVertex();
+            prongIdsD[0] = candD.prong0Id();
+            prongIdsD[1] = candD.prong1Id();
+            prongIdsD[2] = candD.prong2Id();
+            v0_type = V0_type::K0s;
+            d_type = candD.sign() * D_type::Dplus;
+            massV0 = MassK0Short;
+            break;
+
+          case DecayChannel::Xc3055ToDplusLambda:
+            invMassD = hfHelper.invMassDplusToPiKPi(candD);
+            massD = MassDPlus;
+            pVecD = candD.pVector();
+            secondaryVertexD[0] = candD.xSecondaryVertex();
+            secondaryVertexD[1] = candD.ySecondaryVertex();
+            secondaryVertexD[2] = candD.zSecondaryVertex();
+            prongIdsD[0] = candD.prong0Id();
+            prongIdsD[1] = candD.prong1Id();
+            prongIdsD[2] = candD.prong2Id();
+            v0_type = candD.sign() * V0_type::Lambda;
+            d_type = candD.sign() * D_type::Dplus;
+            massV0 = MassLambda0;
+            break;
+
+          default:
+            LOG(warning) << "Decay channel not valid please choose between Ds1ToDstarK0s, DsStar2ToDplusK0s, Xc3055ToDplusLambda";
+            break;
+        } // switch
+      }   // else if
+
+      // Loop on V0 candidates
+      for (const auto& v0 : V0s) {
+        // propagate V0 to primary vertex
+        std::array<float, 3> pVecV0 = {v0.px(), v0.py(), v0.pz()};
+        if (propagateV0toPV) {
+          std::array<float, 3> pVecV0Orig = {v0.px(), v0.py(), v0.pz()};
+          std::array<float, 3> posVecV0 = {v0.x(), v0.y(), v0.z()};
+          gpu::gpustd::array<float, 2> dcaInfo;
+          auto trackParK0 = o2::track::TrackPar(posVecV0, pVecV0Orig, 0, true);
+          trackParK0.setPID(o2::track::PID::K0);
+          trackParK0.setAbsCharge(0);
+          o2::base::Propagator::Instance()->propagateToDCABxByBz({collision.posX(), collision.posY(), collision.posZ()}, trackParK0, 2.f, matCorr, &dcaInfo);
+          getPxPyPz(trackParK0, pVecV0);
+        }
+
```

Comment:
For each V0 candidate we need to test whether it is compatible with a K0S or with a Lambda/AntiLambda. 
To do that, I would add a configurable (or two) with the mass window of compatibility with lambda or K0S. We can also apply other basic cuts in a second iteration (such as cuts on the V0 topology and PID of the daughters)

Then, depending on the compatibility with one or more species, you should set the bit in the `typeV0` (you can use `SETBIT(typeV0, type)`, where `type` can be and integer (`0`, `1`, `2`) or better the element of an `enum` (if you for example extend the existing one adding `AntiLambda`)